src/Tribe/Rewrite.php

```diff
@@ -637,4 +686,231 @@ protected function get_dynamic_matchers( array $query_vars ) {
 	protected function get_post_types() {
 		throw new BadMethodCallException( 'Method get_post_types should be implemented by extending classes.' );
 	}
+
+	/**
+	 * Parses a URL to produce an array of query variables.
+	 *
+	 * Most of this functionality was copied from `WP::parse_request()` method
+	 * with some changes to avoid conflicts and removing non-required behaviors.
+	 *
+	 * @since  TBD
+	 *
+	 * @param string $url              The URLto parse.
+	 * @param array  $extra_query_vars An associative array of extra query vars to use for the parsing. These vars will
+	 *                                 be read before the WordPress defined ones overriding them.
+	 * @param bool   $force Whether to try and use the cache or force a new canonical URL conversion.
+	 *
+	 * @return array An array of query vars, as parsed from the input URL.
+	 */
+	public function parse_request( string $url, array $extra_query_vars = [], $force = false ) {
+		if ( null === $this->rewrite ) {
+			// We re-do this check here as the object might have been initialized before the global rewrite was set.
+			$this->setup();
+		}
+
+		/**
+		 * Allows short-circuiting the URL parsing.
+		 *
+		 * This filter will run before any logic runs, its result will not be cached and this filter will be called on
+		 * each call to this method.
+		 * Returning a non `null` value here will short-circuit this logic.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $query_vars       The parsed query vars array.
+		 * @param array  $extra_query_vars An associative array of extra query vars that will be processed before the
+		 *                                 WordPress defined ones.
+		 * @param string $url              The URL to parse.
+		 */
+		$parsed = apply_filters( 'tribe_rewrite_pre_parse_query_vars', null, $extra_query_vars, $url );
+		if ( null !== $parsed ) {
+			return $parsed;
+		}
+
+		if ( ! $force ) {
+			$this->warmup_cache(
+				'parse_request',
+				WEEK_IN_SECONDS,
+				Listener::TRIGGER_GENERATE_REWRITE_RULES
+			);
+			if ( isset( $this->parse_request_cache[ $url ] ) ) {
+				return $this->parse_request_cache[ $url ];
+			}
+		}
+
+		$query_vars           = [];
+		$post_type_query_vars = [];
+		$perma_query_vars = [];
+		$url_path             = parse_url( $url, PHP_URL_PATH );
+		if ( empty( $url_path ) ) {
+			$url_path = '/';
+		}
+		// Look for matches, removing leading `/` char.
+		$request_match = ltrim( $url_path, '/' );
+		$url_query = parse_url( $url, PHP_URL_QUERY );
+		parse_str( $url_query, $url_query_vars );
+
+		// Fetch the rewrite rules.
+		$rewrite_rules = $this->rewrite->wp_rewrite_rules();
+		$matched_rule = false;
+
+		if ( ! empty( $rewrite_rules ) ) {
+			foreach ( (array) $rewrite_rules as $match => $query ) {
+				$matches_regex = preg_match( "#^$match#", $request_match, $matches )
+				                 || preg_match( "#^$match#", urldecode( $request_match ), $matches );
+
+				if ( $matches_regex ) {
```

Comment:
Perhaps reduce a level of nesting by:

```php
if ( ! $matches_regex ) {
  continue;
}
```

_Perhaps this was copypasta from WP::parse_request()?_